migrator/migrations/m_192_to_m_193_report_config_migrations/migration_impl.go

```diff
@@ -0,0 +1,310 @@
+		for rows.Next() {
+			// convert to report config proto
+			var reportConfig *updatedSchema.ReportConfigurations
+			if err = query.ScanRows(rows, &reportConfig); err != nil {
+				return errors.Wrapf(err, "failed to scan rows")
+			}
+			reportConfigProto, err := updatedSchema.ConvertReportConfigurationToProto(reportConfig)
+			if err != nil {
+				return errors.Wrapf(err, "failed to convert report config  %+v to proto", reportConfigProto)
+			}
+			// skip if version=0 and scope id is not nil since it is v2 config created in tech preview
+			if reportConfigProto.Version == 2 || (reportConfigProto.Version == 0 && reportConfigProto.GetResourceScope() != nil) {
+				reportConfigProto.Version = 2
+				// convert report config proto back to gorm model
+				convertedGormConfig, err := updatedSchema.ConvertReportConfigurationFromProto(reportConfigProto)
+				if err != nil {
+					return errors.Wrapf(err, "failed to convert report config from proto %+v", reportConfigProto)
+				}
+				convertedReportConfigs = append(convertedReportConfigs, convertedGormConfig)
+
+				continue
+			}
+
+			//if deterministic id exists no need to copy the config
+			//since getMigratedReportConfigIfExsists only reads data from older migration, no need to write a new tx
+			migrated, data, err := getMigratedReportConfigIfExsists(reportConfigProto.GetId(), db, database.DBCtx)
+			if err != nil {
+				return errors.Wrapf(err, "failed to query report config with id %s", reportConfigProto.GetId())
+			}
+			if migrated {
+				if reflect.DeepEqual(reportConfig, data) {
+					log.Infof("Old v1 report config proto %+v is different from v2 copy", data)
+				}
+				continue
+			}
+
+			//since checkifNotifierExsists only reads data from older migration, no need to write a new tx
+			notifierFound, err := checkifNotifierExsists(reportConfigProto.GetEmailConfig().GetNotifierId(), db, database.DBCtx)
+			if err != nil {
+				return errors.Wrapf(err, "failed to query notifier with id %s", reportConfigProto.GetEmailConfig().GetNotifierId())
+			}
+
+			if !notifierFound {
+				log.Infof("Did not find notifier with id %s attached to report config %+v", reportConfigProto.GetEmailConfig().GetNotifierId(),
+					reportConfigProto)
+				log.Infof("Did not find notifier with id %s.converted configs %+v", reportConfigProto.GetEmailConfig().GetNotifierId(),
+					convertedReportConfigs)
+				continue
+			}
+
+			// create v2 report config from v1
+			newConfig := createV2reportConfig(reportConfigProto)
+
+			// create notifier config for report_configuration_notifier table
+			notifierConfig := createNotifier(reportConfigProto)
+			newConfig.Notifiers = append(newConfig.Notifiers, notifierConfig)
+
+			// add notifier to notifier_configurations_notifiers
+			reportNotifierGorm, err := updatedSchema.ConvertNotifierConfigurationFromProto(notifierConfig, 0, newConfig.GetId())
+			if err != nil {
+				log.Errorf("failed to convert notifier config from proto %+v", notifierConfig, err)
```

Comment:
nit : Error message can be "failed to convert notifier config from proto to gorm : %+v"